### PR TITLE
util/color/predefinedcolorpalettes: Add Mixxx Track Color Palette

### PR DESCRIPTION
**Pull request by Holzhaus**
The current default track color palette is "Mixxx Hotcue Color Palette", which is confusing and doesn't really fit well for track colors. The "Serato DJ Pro Track Color Palette" works much better.

Originally I wanted to make a PR that switches the default track color palette to that one, but on second thought using Serato's palette in Mixxx by default is weird and looks unprofessional IMHO.

So here's my lazy attempt at making a Mixxx Track Color palette:

colorpalette

**EDIT:** New version:
90941814-795ef800-e413-11ea-9474-1db889c0e69d

### CODE REVIEW

**Review by ronso0 — COMMENTED** [review]
ahm, I tested it locally and I think this 'works' for me, though I don't see the difference between old/new in the first post.? I'll give this a thorough test on monday/tuesday and report here.

I am a bit reserved here since I feel the track colors with the current opacity are not quite WYSIWYG, for example most yellow-ish tones appear like brown or green. But we may improve that experience when we add the track colors to the decks (soon, after #2913 and other PRs I'd like to finish).

**Comment by Holzhaus** [comment]
> I don't see the difference between old/new in the first post.?

I slightly modified colors in column 1, rows 2 and 4.

> I am a bit reserved here since I feel the track colors with the current opacity are not quite WYSIWYG, for example most yellow-ish tones appear like brown or green.

Not if you use the color column instead of just the row colors. But yes, Track colors in decks will improve it further.

**Comment by ronso0** [comment]
:partying_face: right, the color column (where it started..)

interesting to see how colors that seem close in the color column are _not_ when looking at the colored rows..
image

**Comment by ronso0** [comment]
I think this palette is a good start.
Though I'd tweak that yellow/green (n°5) a bit to separate it more from that lime green and pure yellow.
Also, I'd put all pastel colors at the end like this:
image

**Comment by daschuer** [comment]
Nice, I like that.

File: src/util/color/predefinedcolorpalettes.cpp [hunk 1]
```diff
@@ -182,6 +203,31 @@ const ColorPalette PredefinedColorPalettes::kSeratoDJProHotcueColorPalette =
                 },
                 QList<int>{0, 2, 12, 3, 6, 15, 9, 14});
 
+const ColorPalette PredefinedColorPalettes::kMixxxTrackColorPalette =
+        ColorPalette(
+                QStringLiteral("Mixxx Track Colors"),
+                QList<mixxx::RgbColor>{
```
Comment (uklotzde) [inline src/util/color/predefinedcolorpalettes.cpp:209]: Is `QList<mixxx::RgbColor>` actually needed here? I don't need to know the actual type here, the initializer list is fine.

**Review by uklotzde — APPROVED** [review]

**Comment by uklotzde** [comment]
LGTM. Waiting for CI.

**Pull request merged**